src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleDescriptorTest.cs

```diff
@@ -8,70 +10,56 @@ namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules
     public class ConsensusRuleDescriptorTest
     {
         [Fact]
-        public void Constructor_ConsensusRuleWithValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithMempoolAndExecutionAndValidationAttribute_CannotSkipValidation()
         {
-            var rule = new TestConsensusRuleWithValidationAttribute("myRule");
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithValidationExecutionMempoolAttributes());
 
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            descriptor.Rule.Should().BeOfType<RuleWithValidationExecutionMempoolAttributes>();
+            descriptor.RuleAttributes.Should().HaveCount(3);
+            descriptor.RuleAttributes.Should().Contain(x => x is MempoolRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ExecutionRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ValidationRuleAttribute);
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithValidationAttribute).RuleName);
-
-            Assert.Equal(3, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
-            Assert.True(descriptor.Attributes[2] is ValidationRuleAttribute);
-
-            Assert.False(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeFalse();
         }
 
         [Fact]
-        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithOptionalValidationAttribute_CanSkipValidation()
         {
-            var rule = new TestConsensusRuleWithOptionalValidationAttribute("myRule");
-
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithOptionalValidationAttribute());
```

Comment:
This rule seems different from the original `TestConsensusRuleWithOptionalValidationAttribute` which has 3 attributes. Is that intentional?

Reply:
Yes - the test was only concerned with the validation attribute so the other two were superfluous to the test

Reply:
and you have a test that tests that you have the right number of attributes when they're > 1.